mcs/class/System/Mono.Net.Security/MonoTlsProviderFactory.cs

```diff
@@ -160,9 +162,17 @@ static void InitializeProviderRegistration ()
 					return;
 				providerRegistration = new Dictionary<string,string> ();
 				providerRegistration.Add ("legacy", "Mono.Net.Security.LegacyTlsProvider");
-				providerRegistration.Add ("default", "Mono.Net.Security.LegacyTlsProvider");
+			
+				if (Platform.IsMacOS)
+					providerRegistration.Add ("default", "Mono.AppleTls.AppleTlsProvider");
+				else
+					providerRegistration.Add ("default", "Mono.Net.Security.LegacyTlsProvider");
+
 				if (IsBtlsSupported ())
 					providerRegistration.Add ("btls", "Mono.Btls.MonoBtlsProvider");
+			
+				providerRegistration.Add ("apple", "Mono.AppleTls.AppleTlsProvider");
```

Comment:
We didn't have one around the BTLS registration below so I followed suit.

I can add (one or both) if needed.